code/src/test/kotlin/loggerTests/LoggerTest.kt

```diff
@@ -0,0 +1,76 @@
+package loggerTests
+import logger.*
+import org.junit.jupiter.api.Assertions.assertEquals
+
+class TokenTest {
+
+    @org.junit.jupiter.api.Test
+    fun testTypeError() {
+        val error = TypeError(4, 10, "var b = 4 < \"5\"", "<", listOf("num", "txt", "bool"))
+        val logger = TestLogger()
+        logger.logCompilationError(error)
+        assertEquals("- ERROR: TYPE-ERROR found at line 4 index 10:\n" +
+                             " | var b = 4 < \"5\"\n" +
+                             " |           ^--\n" +
+                             " | Function '<' not defined for types: 'num', 'txt' and 'bool'.\n" +
+                             " | -->help: Try casting your types to match eachother.\n", logger.buffer.toString())
+    }
+
+    @org.junit.jupiter.api.Test
+    fun testUndeclaredError() {
+        val error = UndeclaredError(12, 8, "txt s = textString", "textString")
+        val logger = TestLogger()
+        logger.logCompilationError(error)
+        assertEquals("- ERROR: UNDECLARED-ERROR found at line 12 index 8:\n" +
```

Comment:
make indents the same. move the first string (in each of these) to the next line.